### PR TITLE
Cr 1606 number in household

### PR DESCRIPTION
**Pull request by Paul-Joel**
# Motivation and Context
Additional validation on 'Number in Household'

# What has changed
Text change in error message
Add upper limit to 'number in household'
Ensure that value of '0' doesn't cause an error (was causing an error for large-print)
Adds additional unit tests to cover new variations

No Cucumber updates required

# How to test?
Check on household paper-questionnaire fulfilment that number over 30 gives an error
Check on household paper-questionnaire fulfilment that entering '0' in household works when selecting 'large print'

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-1606

### CODE REVIEW

File: tests/unit/__init__.py [hunk 1]
```diff
@@ -1332,14 +1332,19 @@ def setUp(self):
         self.content_request_questionnaire_people_in_household_title_en = 'How many people are in your household?'
         self.content_request_questionnaire_people_in_household_error_empty_en = \
             'Enter the number of people in your household'
-        self.content_request_questionnaire_people_in_household_error_nan_en = 'Enter a numeral'
+        self.content_request_questionnaire_people_in_household_error_nan_en = 'Enter a number'
+        self.content_request_questionnaire_people_in_household_error_number_less_en = \
+            'Enter a number less than 30'
```
Comment (cookel2) [inline tests/unit/__init__.py:1337]: similarly here

**Review by cookel2 — CHANGES_REQUESTED** [review]

File: app/utils.py [hunk 2]
```diff
@@ -369,13 +369,39 @@ def validate_number_of_people(request, data, display_region, request_type):
...
                                                                        'ERROR', 'NUMBER_OF_PEOPLE_ERROR',
-                                                                       'number_of_people_continuation'))
+                                                                       'number_of_people_continuation_low'))
                 number_of_people_valid = False
 
+            elif int(data.get('number_of_people')) > 30:
+                logger.info('number_of_people continuation greater than 30', client_ip=request['client_ip'])
+                if display_region == 'cy':
+                    # TODO Add Welsh Translation
+                    flash(request, FlashMessage.generate_flash_message('Enter a number less than 30',
```
[the remark this answers is not in the document]
Comment (Paul-Joel) [reply app/utils.py]: Changed flash message text to 31 throughout

File: tests/unit/__init__.py (hunk 1 above)
Comment (Paul-Joel) [reply tests/unit/__init__.py]: Changed flash message text to 31 throughout

**Review by andrewjohnysons — APPROVED** [review]
Approved pending fixes for Ella (less than 31 fix)

**Review by cookel2 — APPROVED** [review]
approved by me too

**Pull request merged**